src/group/mod.rs

```diff
@@ -1,71 +1,175 @@
+            ProposalStore::default(),
+        )?;
+        Ok(Self {
+            group_info,
+            public_group,
+            backend,
+        })
+    }
 
-    pub fn group_info(&self) -> &GroupInfo {
-        &self.group_info
+    fn backend(&self) -> &OpenMlsRustCrypto {
+        &self.backend
     }
 
-    pub fn members() {}
-}
+    /// Returns a [`ProcessedMessage`] for inspection.
+    pub fn process(
+        self,
+        assisted_message: AssistedMessage,
+    ) -> Result<ProcessedAssistedMessage, ProcessAssistedMessageError> {
+        match assisted_message {
+            AssistedMessage::NonCommit(public_message) => {
+                let processed_message = self
+                    .public_group
+                    .process_message(self.backend(), public_message)?;
+                Ok(ProcessedAssistedMessage::NonCommit(processed_message))
+            }
+            AssistedMessage::Commit(AssistedCommit {
+                commit,
+                assisted_group_info,
+            }) => {
+                // First process the message, then verify that the group info
+                // checks out.
+                let processed_message = self
+                    .public_group
+                    .process_message(self.backend(), commit.clone())?;
+                let sender = processed_message.sender().clone();
+                let confirmation_tag = commit
+                    .confirmation_tag()
+                    .ok_or(LibraryError::custom(
+                        "No confirmation tag in commit after validation.",
+                    ))?
+                    .clone();
+                if let ProcessedMessageContent::StagedCommitMessage(staged_commit) =
+                    processed_message.content()
+                {
+                    let group_info: GroupInfo = self.validate_group_info(
+                        sender,
+                        staged_commit,
+                        confirmation_tag,
+                        assisted_group_info,
+                    )?;
+                    // This is really only relevant for the "Full" group info case above.
+                    if group_info.group_context() != staged_commit.staged_context() {
+                        return Err(ProcessAssistedMessageError::InconsistentGroupContext);
+                    }
+                    Ok(ProcessedAssistedMessage::Commit(
+                        processed_message,
+                        group_info,
+                    ))
+                } else {
+                    Err(ProcessAssistedMessageError::LibraryError(
+                        LibraryError::custom("Mismatching message type."),
+                    ))
+                }
+            }
+        }
+    }
 
-#[derive(
-    Debug, PartialEq, Eq, Clone, Serialize, Deserialize, TlsDeserialize, TlsSerialize, TlsSize,
-)]
-pub struct GroupId;
+    pub fn accept_message(&mut self, processed_assisted_message: ProcessedAssistedMessage) {
```

Comment:
Can you add a comment on what this does?

Reply:
Will do. I'm also not super happy with the name, so if you have a suggestion, I'm all ears.